frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx

```diff
@@ -98,20 +116,47 @@ export default function CourseDescription({ structure }) {
         prepareUserPayload(degree, planner)
       );
       console.log(data);
-      if (!err) setCoursesPathTo({
-        "direct_unlock": data.direct_unlock,
-        "indirect_unlock": data.indirect_unlock
-      });
+      if (!err)
+        setCoursesPathTo({
+          direct_unlock: data.direct_unlock,
+          indirect_unlock: data.indirect_unlock,
+        });
+    };
+
+    const getCourseCapacityById = async (id) => {
+      const [data, err] = await axiosRequest(
+        "get",
+        `https://timetable.csesoc.app/api/terms/${term}/courses/${id}`
+      );
+      console.log(data);
+      if (!err) {
+        getCapacityAndEnrolment(data);
+      }
     };
 
-    setpageLoaded(false);
     if (id === "explore" || id === "search") return;
     if (id) {
       getCourse();
       getPathToCoursesById(id);
+      getCourseCapacityById(id);
     }
+    setPageLoaded(true);
   }, [id]);
 
+  const getCapacityAndEnrolment = (data) => {
+    let enrolmentCapacityData = {
+      enrolments: 0,
+      capacity: 0,
+    };
+    for (let i = 0; i < data.classes.length; i++) {
+      if (data.classes[i].activity === "Lecture" || data.classes[i].activity === "Seminar" || data.classes[i].activity === "Thesis Research") {
+        enrolmentCapacityData.enrolments += data.classes[i].courseEnrolment.enrolments;
+        enrolmentCapacityData.capacity += data.classes[i].courseEnrolment.enrolments;
```

Comment:
Correct me if I'm wrong but wouldn't `enrolmentCapacityData.enrolments` and `enrolmentCapacityData.capacity` end up as the same value after the loop? Since both are adding `data.classes[i].courseEnrolment.enrolments`.

Reply:
Yeah that's a typo. ignore this for now. This is a rough amendment. Need to think about the logic a bit more